### PR TITLE
feat: add --md-only flag

### PR DESCRIPTION
**Pull request by giacomoguiulfo**
Signed-off-by: Giacomo Guiulfo <giacomoguiulfo@gmail.com>

#### What is the purpose of this change?

Add support for a global flag, "md-only". If enabled, only metadata of the files will be uploaded to the cloud backends. Also, a metadata header "Mdonly" will be included so that a cloud backend is aware that metadata-only put(s) were intended.

#### Was the change discussed in an issue or in the forum before?

Partially. This PR represent a subset of the changes discussed in https://forum.rclone.org/t/min-age-and-max-age-support-for-atime-and-ctime/2122/2.

#### Checklist

- [ ] I have read the [contribution guidelines](https://github.com/ncw/rclone/blob/master/CONTRIBUTING.md#submitting-a-pull-request).
- [ ] I have added tests for all changes in this PR if appropriate.
- [ ] I have added documentation for the changes if appropriate.
- [ ] All commit messages are in [house style](https://github.com/ncw/rclone/blob/master/CONTRIBUTING.md#commit-messages).
- [ ] I'm done, this Pull Request is ready for review :-)

### CODE REVIEW

**Comment by ncw** [comment]
What is the purpose of this change?  It seems to upload 0 size objects only?  Why would you want to do that?  Or have I mis-understood - thanks!

**Comment by giacomoguiulfo** [comment]
@ncm Hello Nick. Yes, the purpose of this change is to be able to upload 0 byte objects with their proper metadata attributes (system attributes, xattr, etc.) similarly to how `cp --attributes-only` works. There is a thread which discusses this topic further https://askubuntu.com/questions/56792/how-to-copy-only-file-attributes-metadata-without-actual-content-of-the-file.

I have to admit that clouds don't really leverage anything similar to this as a feature. So far, its purpose is to help us build one of the components we are developing here at Scality for [Zenko](https://github.com/scality/Zenko).

In short, this component will `rclone copy` metadata from a filesystem, and then a s3-like backend will "read" the filesystem using this metadata as reference. This allows you to manage data from the cloud, without having to copy or move it there. You can find more detailed information about this [here](https://github.com/scality/Zenko-RFCs/pull/5).

Let me know your thoughts about it, thanks!

**Comment by giacomoguiulfo** [comment]
Closing this since it's not a general use case.
